test/translation/lua/assignments.lua

```diff
@@ -43,23 +43,23 @@ obj.prop = (obj.prop & arr[(0)+1]);
-do local __TS_obj, __TS_index = getObj(), "prop"; __TS_obj[__TS_index] = (__TS_obj[__TS_index] << (getArr()[(getIndex())+1])); end;
-do local __TS_obj, __TS_index = getObj(), "prop"; __TS_obj[__TS_index] = (__TS_obj[__TS_index] >> (getArr()[(getIndex())+1])); end;
+do local __TS_obj, __TS_index = arr, (getIndex())+1; local __TS_tmp = (__TS_obj[__TS_index]+(1)); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = arr, (getIndex())+1; local __TS_tmp = __TS_obj[__TS_index]; __TS_obj[__TS_index] = (__TS_tmp-(1)); end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index]+(1)); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = __TS_obj[__TS_index]; __TS_obj[__TS_index] = (__TS_tmp+(1)); end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index]-(1)); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = __TS_obj[__TS_index]; __TS_obj[__TS_index] = (__TS_tmp-(1)); end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index]+(getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index]-(getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index]*(getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getArr(), (getIndex())+1; local __TS_tmp = (__TS_obj[__TS_index]/(getObj().prop)); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getArr(), (getIndex())+1; local __TS_tmp = (__TS_obj[__TS_index]%(getObj().prop)); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getArr(), (getIndex())+1; local __TS_tmp = (__TS_obj[__TS_index]^(getObj().prop)); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index] | (getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index] & (getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index] ~ (getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index] << (getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
+do local __TS_obj, __TS_index = getObj(), "prop"; local __TS_tmp = (__TS_obj[__TS_index] >> (getArr()[(getIndex())+1])); __TS_obj[__TS_index] = __TS_tmp; end;
 z = ((function() x = y; return y end)());
 z = ((function() obj.prop = x; return x end)());
```

Comment:
Just a general remark regarding translation tests: We don't really value them that much (and are considering maybe just getting rid of them in favor of functional tests). This under the reasoning that we don't really care that much about what our resulting lua looks like, as long as it works. For now I'm fine with these though since this does ensure the stability of our translation.

Reply:
I agree the translation tests should not be taken as seriously as the functional, but I wouldn't remove them as they force you to pay attention when something changes in the output lua that may have unintended consequences. Perhaps something the functional tests miss, or maybe the result is still correct but the new generated lua is terribly inefficient. But yes, I think they should only be a back-up to the functional tests, not a replacement.